### ISSUE TITLE
computeQuickXorHash generates a different hash when files are > 64Kb

### ISSUE THREAD

**Issue by abraunegg** [abraunegg/onedrive#23]
When looking at the solution for #2, it has been identified that the function which generates the quickXorHash hash for comparison with what is returned by OneDrive, generates a different value to what is expected when the file size is > 64Kb.

When the file is under 64Kb, the quickXorHash that is generated matches what is returned from OneDrive.

Simply 'touching' a local file > 64Kb will trigger this hash difference:

```
The file last modified time has changed
file mtime = 2018-Jun-16 00:52:06Z
item mtime = 2018-Jun-16 00:51:30Z
item hash = tJIwyksN/FCJfJML1uY05oBbHXY=
file hash = tJIwyksN/FBrg2z0KRnL+6hbHXY=
The file content has changed
Uploading file hcw85prop.ax ... done.
```

In reality if a local file has been modified, the content 'may' have changed, thus the modified file will be uploaded as expected, however when simply 'touching' the file this should not occur (upload modified file due to contents changing - when it hasnt).

Seeking some assistance to look at qxor.d to work out where things are potentially going wrong as it is not immediately obvious.

OneDrive API developer documentation around quickXorHash is here:

https://docs.microsoft.com/en-us/onedrive/developer/code-snippets/quickxorhash

**Comment by abraunegg** [comment]
This issue is probably one of the causes of the 'ETag does not match current item's value' issue that has plagued this client since the early days.

**Comment by robertjf** [comment]
looking at the reference implementation (https://github.com/OneDrive/onedrive-api-docs/blob/live/docs/code-snippets/quickxorhash.md), there are a few slight differences - will see how I go if I have time this morning

**Comment by abraunegg** [comment]
Thanks - any assistance is greatly appreciated

**Comment by robertjf** [comment]
well, I got this one licked.  PR incoming, just got to fix up my repository first.
Actually, might be easier if I just put the necessary changes in here:

https://github.com/abraunegg/onedrive/blob/c0d2874acee4bf86822a2295f1d76e6a01c44f6c/src/qxor.d#L35-L37

change 
```
				byte low = cast(byte) (bitsInVectorCell - vectorOffset);

				byte xoredByte = 0;
```
to
```
				ubyte low = cast(ubyte) (bitsInVectorCell - vectorOffset);

				ubyte xoredByte = 0;
```

that's it.  problem solved :)

I also made a few other minor cosmetic changes to my local copy, but they shouldn't have any impact.
